### PR TITLE
3134/description field to collection model

### LINKED ISSUE #3134: Add description field to collection model
### What I'm trying to achieve
To add a description to a collection.

### Describe a proposed solution
Add `description` field to `Collection` model.

### TODO 
- [ ] Add description field to `Collection` model
- [ ] Add description field to `CollectionTranslation` model
- [ ] Adding migrations
- [ ] Add `description` to Collection's forms & templates in the dashboard
- [ ] Adding `description` field to Collection's GraphQL APIs (mutations and types)
- [ ] Adjust "SEO Widget" logic on the dashboard, on Collection Add/Edit page (description should be used as a placeholder if seo_description is missing)
- [ ] Adjust tests for all of the above

### PR DESCRIPTION
**Pull request by fowczarek**
I want to merge this change because resolve #3134

### Pull Request Checklist

1. [x] Privileged views and APIs are guarded by proper permission checks.
1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Database queries are optimized and the number of queries is constant.
1. [x] The changes are tested.
1. [x] The code is documented (docstrings, project documentation).
1. [x] GraphQL schema and type definitions are up to date.

### CODE REVIEW

File: saleor/product/models.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from django.conf import settings
 from django.contrib.postgres.fields import HStoreField
 from django.core.exceptions import ValidationError
-from django.core.validators import MinValueValidator
+from django.core.validators import MaxLengthValidator, MinValueValidator
```
Comment (Pacu2) [inline saleor/product/models.py]: Why is this validator needed?

File: saleor/dashboard/collection/forms.py [hunk 2]
```diff
@@ -22,17 +22,21 @@ class Meta:
         labels = {
             'name': pgettext_lazy('Item name', 'Name'),
             'background_image': pgettext_lazy(
-                'Products selection',
+                'Background Image of collection',
```
Comment (maarcingebala) [inline saleor/dashboard/collection/forms.py]: Should be "Background image of a collection".

File: saleor/product/models.py (hunk 1 above)
Comment (maarcingebala) [reply saleor/product/models.py]: Seems unused, should be deleted.

File: saleor/core/utils/random_data.py [hunk 3]
```diff
@@ -47,10 +47,12 @@
 COLLECTIONS_SCHEMA = [
     {
         'name': 'Summer collection',
-        'image_name': 'summer.jpg'},
+        'image_name': 'summer.jpg',
+        'description': 'Description of a summer collection.'},
```
Comment (Pacu2) [inline saleor/core/utils/random_data.py]: Could you get a nicer copy for this? Right now populatedb creates real-like data

File: saleor/product/models.py [hunk 4]
```diff
@@ -454,6 +453,7 @@ class Collection(SeoModel):
     background_image = VersatileImageField(
         upload_to='collection-backgrounds', blank=True, null=True)
     is_published = models.BooleanField(default=False)
+    description = models.TextField(blank=True, null=True)
```
Comment (maarcingebala) [inline saleor/product/models.py]: In the Category model, this field is not nullable (while it allows for blank values). Can we keep consistency and have the same signature here? This affects `schema.graphql` so we'd have to regenerate it. For migrations, we could just change the `0078_auto_20181119_0535.py` accordingly.

**Pull request merged**